server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -1500,6 +1512,11 @@ private static void failIfOverShardCountLimit(ClusterService clusterService, int
         }
     }
 
+    /**
+     * {@link ActionListener} suitable for aggregating cross-cluster responses.
```

Comment:
Maybe "collecting" instead of "aggregating"? The latter word has different meanings in Elasticsearch.